mlflow/tracking/_model_registry/fluent.py

```diff
@@ -675,6 +684,34 @@ def load_prompt(
     if run := active_run():
         client.log_prompt(run.info.run_id, f"prompts:/{prompt.name}/{prompt.version}")
 
+    if link_to_model:
+        model_id = model_id or get_active_model_id()
+        if model_id is not None:
+            # Run linking in background thread to avoid blocking prompt loading. Prompt linking
+            # is not critical for the user's workflow (if the prompt fails to link, the user's
+            # workflow is minorly affected), so we handle it asynchronously and gracefully
+            # handle any failures without impacting the core prompt loading functionality.
+
+            def _link_prompt_async():
+                try:
+                    client.link_prompt_version_to_model(
+                        name=prompt.name,
+                        version=prompt.version,
+                        model_id=model_id,
+                    )
+                except Exception:
+                    # NB: We should still load the prompt even if linking fails, since the prompt
+                    # is critical to the caller's application logic
+                    _logger.warn(
+                        f"Failed to link prompt '{prompt.name}' version '{prompt.version}'"
+                        f" to model '{model_id}'.",
+                        exc_info=True,
+                    )
+
+            # Start linking in background - don't wait for completion
+            link_thread = threading.Thread(target=_link_prompt_async)
```

Comment:
```suggestion
            link_thread = threading.Thread(target=_link_prompt_async, name="...")
```

Let's add a name. This allows us to look up this thread using `threading.enumerate` and join it.